macOS/DuckDuckGo/Autoconsent/AutoconsentDailyStats.swift

```diff
@@ -0,0 +1,147 @@
+    private let featureFlagger: FeatureFlagger
+    private let currentDateProvider: () -> Date
+    private let queue: DispatchQueue
+    private let firePixel: (AutoconsentPixel, PixelKit.Frequency) -> Void
+
+    init(keyValueStore: ThrowingKeyValueStoring,
+         featureFlagger: FeatureFlagger,
+         currentDateProvider: @escaping () -> Date = { Date() },
+         queue: DispatchQueue = DispatchQueue(label: "com.duckduckgo.autoconsent.stats"),
+         firePixel: @escaping (AutoconsentPixel, PixelKit.Frequency) -> Void = { PixelKit.fire($0, frequency: $1) }
+    ) {
+        self.keyValueStore = keyValueStore
+        self.featureFlagger = featureFlagger
+        self.currentDateProvider = currentDateProvider
+        self.queue = queue
+        self.firePixel = firePixel
+    }
+
+    private var dailyStats: [Date: Int] {
+        get {
+            do {
+                guard let data = try keyValueStore.object(forKey: Constants.statsKey) as? Data,
+                      let stats = try? JSONDecoder().decode(Stats.self, from: data) else {
+                    return [:]
+                }
+                return stats.counts
+            } catch {
+                Logger.autoconsent.error("Failed to read daily stats: \(error.localizedDescription)")
+                return [:]
+            }
+        }
+        set {
+            do {
+                let stats = Stats(counts: newValue)
+                let data = try JSONEncoder().encode(stats)
+                try keyValueStore.set(data, forKey: Constants.statsKey)
+            } catch {
+                Logger.autoconsent.error("Failed to save daily stats: \(error.localizedDescription)")
+            }
+        }
+    }
+
+    func incrementPopupCount() {
+        guard featureFlagger.isFeatureOn(.cpmCountPixel) else { return }
+        queue.async {
+            let today = self.startOfDay(for: self.currentDateProvider())
+            self.dailyStats[today, default: 0] += 1
+        }
+    }
+
+    private func cleanOldStats() {
+        let today = self.startOfDay(for: self.currentDateProvider())
+
+        var stats = self.dailyStats
+        stats = stats.filter { date, _ in
+            let days = Calendar.current.dateComponents([.day], from: date, to: today).day ?? 0
+            return days <= Constants.maxDaysToKeep
+        }
+        self.dailyStats = stats
+    }
+
+    func sendDailyPixelIfNeeded() {
+        guard featureFlagger.isFeatureOn(.cpmCountPixel) else { return }
+        queue.async {
+            let today = self.startOfDay(for: self.currentDateProvider())
+
+            // Get stats for last 10 days
+            var params: [String: String] = [:]
+            var day1Count = 0
+            var day2Count = 0
+            var day5Count = 0
+            var day10Count = 0
+            for daysAgo in 0..<Constants.maxDaysToKeep {
+                if let date = Calendar.current.date(byAdding: .day, value: -(daysAgo + 1), to: today) {
```

Comment:
Nitpick, but we already have more readable date helpers like `daysAgo(_ days: Int) -> Date`